src/clj_kondo/impl/analysis/java.clj

```diff
@@ -47,7 +47,7 @@
   (-> ctx :config ))
 
 (defn analyze-class-defs? [ctx]
-  (:analyze-java-class-defs? ctx))
+  (and (:analysis ctx) (:analyze-java-class-defs? ctx)))
```

Comment:
Why is this necessary? How can analyze-java-class-defs? be true while analysis is not true?